README.md

```diff
@@ -2310,6 +2310,39 @@ val d: Int < IO =
     a.map(_.pending)
 ```
 
+### Barrier: Multi-party Rendezvous
+
+The `Barrier` effect provides a synchronization primitive that allows a fixed number of parties to wait for each other to reach a common point of execution. It's particularly useful in scenarios where multiple fibers need to synchronize their progress.
+
+```scala
+import kyo.*
+
+// Initialize a barrier for 3 parties
+val a: Barrier < IO =
+    Barrier.init(3)
+
+// Wait for the barrier to be released
+val b: Unit < Async =
+    a.map(_.await)
+
+// Get the number of parties still waiting
+val c: Int < IO =
+    a.map(_.pending)
+
+// Example usage with multiple fibers
+val d: Unit < Async =
+    for
+        barrier <- Barrier.init(3)
+        _       <- Async.parallel(
+                     barrier.await,
```

Comment:
I presume you would map an existing fiber to barrier.await? Might be good to show that?